File: app/src/main/java/com/fuck/viewtest/menu/view_/anim/animator/evaluator/PointEvaluator.java
package com.fuck.viewtest.menu.view_.anim.animator.evaluator;

import android.animation.TypeEvaluator;

import com.fuck.viewtest.menu.view_.anim.animator.MyPoint;

public class PointEvaluator implements TypeEvaluator<MyPoint> {
    @Override
    public MyPoint evaluate(float fraction, MyPoint startValue, MyPoint endValue) {

        //起始
        float x1 = startValue.getX();
        float y1 = startValue.getY();

        //结束
        float x2 = endValue.getX();
        float y2 = endValue.getY();

        //当前
        float x = x1 + fraction * (x2 - x1);
        float y = y1 + fraction * (y2 - y1);

        return new MyPoint(x, y);
    }
}

File: app/src/main/java/com/fuck/viewtest/menu/anim/animator/PointEvaluator.kt
package com.fuck.viewtest.menu.anim.animator

import android.animation.TypeEvaluator
import com.fuck.viewtest.menu.anim.animator.Point


class PointEvaluator:TypeEvaluator<Point> {

    override fun evaluate(pFraction: Float, pStartValue: Point, pEndValue: Point): Point {
        //解构声明
        val (x1, y1) = pStartValue
        //解构声明
        val (x2, y2) = pEndValue
        //当前进度
        val x = x1 + pFraction * (x2 - x1)
        //当前进度
        val y = y1 + pFraction * (y2 - y1)
        //动画当前值
        return Point(x, y)
    }
}